addons/crm_livechat/models/chatbot_script_step.py

```diff
@@ -1,23 +1,35 @@
-# -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import _, models, fields
+from ast import literal_eval
+
+from odoo import models, fields
+from odoo.osv import expression
 
 
 class ChatbotScriptStep(models.Model):
     _inherit = 'chatbot.script.step'
 
     step_type = fields.Selection(
-        selection_add=[('create_lead', 'Create Lead')], ondelete={'create_lead': 'cascade'})
+        selection_add=[
+            ("create_lead", "Create Lead"),
+            ("create_lead_and_forward", "Create Lead & Forward"),
+        ],
+        ondelete={"create_lead": "cascade", "create_lead_and_forward": "cascade"},
+    )
     crm_team_id = fields.Many2one(
-        'crm.team', string='Sales Team', ondelete='set null',
+        'crm.team', string='Sales Team', ondelete='set null', index="btree_not_null",
         help="Used in combination with 'create_lead' step type in order to automatically "
              "assign the created lead/opportunity to the defined team")
 
+    def _compute_is_forward_operator(self):
+        super()._compute_is_forward_operator()
+        self.filtered(lambda s: s.step_type == "create_lead_and_forward").is_forward_operator = True
+
     def _chatbot_crm_prepare_lead_values(self, discuss_channel, description):
         return {
+            "company_id": self.crm_team_id.company_id.id,
```

Comment:
Don't do that, this leads to multicompany issues. We are going to lessen number of "forced" company_id, so why adding it here ? Seems irrelevant to the task anyway, isn't it ?